AppAuthMigration doc update (#2268)
Azure/azure-docs-sdk-dotnet: api/overview/azure/latest/app-auth-migration.md

|Auth Provider|AppAuthentication<br>Connection String|Azure.Identity<br>TokenCredential|
|---------|---------|---------|

> |Default / environment-based | Default - no connection string used | new DefaultAzureCredential()*|

Reviewer: ```suggestion
|(Default) environment-based | Default - no connection string used | new DefaultAzureCredential()*|
```

> [!NOTE]
> `*` Authentication providers and order is different between [AzureServiceTokenProvider](https://github.com/Azure/azure-sdk-for-net/blob/7d23a9d912da40baeebee1125eb5ebefa78449a2/sdk/mgmtcommon/AppAuthentication/Azure.Services.AppAuthentication/AzureServiceTokenProvider.cs#L104) and [DefaultAzureCredential](https://docs.microsoft.com/dotnet/api/overview/azure/identity-readme#defaultazurecredential)<br>

> > `**` Need to set [environment variables](https://learn.microsoft.com/en-us/dotnet/api/overview/azure/identity-readme#environment-variables)

Reviewer: ```suggestion
> `**` Need to set [environment variables](https://learn.microsoft.com/dotnet/api/overview/azure/identity-readme#environment-variables)
```

-   System.Data.SqlClient.SqlAuthenticationProvider implementation
    (SqlAppAuthenticationProvider)

>     -   For Microsoft.Data.SqlClient, see [Active Directory Default authentication](https://learn.microsoft.com/en-us/sql/connect/ado-net/sql/azure-active-directory-authentication#using-active-directory-default-authentication). This authentication mode provides similar functionality where DefaultAzureCredential is used to obtain access token for authentication to SQL instance

Reviewer: ```suggestion
    -   For Microsoft.Data.SqlClient, see [Active Directory Default authentication](https://learn.microsoft.com/sql/connect/ado-net/sql/azure-active-directory-authentication#using-active-directory-default-authentication). This authentication mode provides similar functionality where DefaultAzureCredential is used to obtain access token for authentication to SQL instance
```

>     -   For Microsoft.Data.SqlClient, see [Active Directory Default authentication](https://learn.microsoft.com/en-us/sql/connect/ado-net/sql/azure-active-directory-authentication#using-active-directory-default-authentication). This authentication mode provides similar functionality where DefaultAzureCredential is used to obtain access token for authentication to SQL instance

Reviewer 2: ```suggestion
    -   For Microsoft.Data.SqlClient, see [Active Directory Default authentication](https://learn.microsoft.com/sql/connect/ado-net/sql/azure-active-directory-authentication#using-active-directory-default-authentication). This authentication mode provides similar functionality where DefaultAzureCredential is used to obtain access token for authentication to SQL instances.
```

Reviewer 2: Anais caught the culture slug already, but it seemed weird to leave it out of the suggestion...

-   Directly use certificates in cert store as client credential (using subject name or thumbprint identifier)

-   Directly use certificates in Key Vault as client credential (using Key Vault certificate secret identifier)

-   Change authentication provider with environment config (i.e. connection strings in AppAuthentication)

>     -   Limited support in Azure.Identity with DefaultAzureCredential and EnvironmentCredential, see [environment variables](https://learn.microsoft.com/en-us/dotnet/api/overview/azure/identity-readme#environment-variables)

Reviewer: ```suggestion
    -   Limited support in Azure.Identity with DefaultAzureCredential and EnvironmentCredential, see [environment variables](https://learn.microsoft.com/dotnet/api/overview/azure/identity-readme#environment-variables)
```

-   Determine auth provider and identity used for environment-based authentication (i.e. AzureServiceTokenProvider.PrincipalUsed property)

>     - Can determine auth provider used with DefaultAzureCredential in Azure.Identity using [AzureEventSourceListener](https://learn.microsoft.com/en-us/dotnet/api/azure.core.diagnostics.azureeventsourcelistener)

Reviewer: ```suggestion
    - Can determine auth provider used with DefaultAzureCredential in Azure.Identity using [AzureEventSourceListener](https://learn.microsoft.com/dotnet/api/azure.core.diagnostics.azureeventsourcelistener)
```

);
```

> [!NOTE]

> More information on the `.default` scope can be found [here](https://learn.microsoft.com/azure/active-directory/develop/scopes-oidc#the-default-scope).

Reviewer: ```suggestion
> More information on the `.default` scope can be found [here](https://learn.microsoft.com/azure/active-directory/develop/scopes-oidc#the-default-scope).
```
